packages/chevrotain/src/parse/parser/traits/recognizer_engine.ts

```diff
@@ -210,35 +213,30 @@ export class RecognizerEngine {
     this.shortRuleNameToFull[shortName] = ruleName
     this.fullRuleNameToShort[ruleName] = shortName
 
-    function invokeRuleWithTry(this: MixedInParser, args: any[]) {
+    function invokeRuleWithTry(this: MixedInParser, ...args: ARGS[]): R {
       try {
+        this.ruleInvocationStateUpdate(shortName, ruleName, this.subruleIdx)
+        this.subruleIdx = 0
```

Comment:
why do we need to reset `this.subruleIdx = 0' ? won't the next `SUBRULE` invocation take care of it?

Reply:
`invokeRuleWithTry` is a hotspot, so we want to avoid any none mandatory statements.

Reply:
I believe there is a `reset` method on this class/trait where it should be reset to 0 instead.

Reply:
I've got it in init, it's not strictly required in reset, as top-level rule calls should work with any idx right?

Reply:
Yeah, good point, it may not be **strictly** required.
I may add it **later** though to be explicit and be more consistent between different invocations (inputs) on the same parser instance.